src/Microsoft.OData.Core/UrlValidation/ODataUrlValidationMessage.cs

```diff
@@ -0,0 +1,88 @@
+﻿//---------------------------------------------------------------------
+// <copyright file="ODataUrlValidationMessage.cs" company="Microsoft">
+//      Copyright (C) Microsoft Corporation. All rights reserved. See License.txt in the project root for license information.
+// </copyright>
+//---------------------------------------------------------------------
+
+using System.Collections.Generic;
+
+namespace Microsoft.OData.UriParser.Validation
+{
+    /// <summary>
+    /// Severity of an OData Url Validation Message
+    /// </summary>
+    public enum Severity
+    {
+        /// <summary>
+        /// Default / Undefined error Level
+        /// </summary>
+        Undefined = 0,
+
+        /// <summary>
+        /// Informational Message
+        /// </summary>
+        Message = 1,
+
+        /// <summary>
+        /// Warning
+        /// </summary>
+        Warning = 2,
+
+        /// <summary>
+        /// Error
+        /// </summary>
+        Error = 3,
+    }
+
+    /// <summary>
+    /// A class representing a message encountered while validating an OData Url.
+    /// </summary>
+    public class ODataUrlValidationMessage
+    {
+        private Dictionary<string, object> additionalInfo;
+
+        /// <summary>
+        /// Construct an instance of an ODataUrlValidationMessage given a message code, message, and severity.
+        /// </summary>
+        /// <param name="code">The message code of the error.</param>
+        /// <param name="message">The message.</param>
+        /// <param name="severity">The severity of the message.</param>
+        public ODataUrlValidationMessage(string code, string message, Severity severity)

```

Comment:
Consider add a overload without Severity?

Reply:
I think severity should be explicit (not sure what we'd make the default). We can add an overload in the future if we decide to support a default.

---
In reply to: [434871734](https://github.com/OData/odata.net/pull/1769#discussion_r434871734) [](ancestors = 434871734)